### PR TITLE
Add parse member function

### PR DESCRIPTION
**Pull request by miguel-vila**
Adds a function to parse shape ids with members

### CODE REVIEW

**Review by daddykotex — APPROVED** [review]
That makes sense too me. Plus TIL that you can deconstruct an `Array` directly like that!

**Comment by daddykotex** [comment]
Your branch is outdated though

File: modules/core/src/smithy4s/ShapeId.scala [hunk 1]
```diff
@@ -40,12 +40,29 @@ object ShapeId extends ShapeTag.Has[ShapeId] { self =>
...
-  final case class Member(shapeId: ShapeId, member: String)
-
   val id: ShapeId = ShapeId("smithy4s", "ShapeId")
   lazy val schema =
     Schema.string.refined[ShapeId](IdRef()).withId(id)
 
+  final case class Member(shapeId: ShapeId, member: String)
+
+  object Member {
+    def parse(string: String): Option[Member] = {
```
Comment (miguel-vila) [inline modules/core/src/smithy4s/ShapeId.scala:50]: think if you call `ShapeId.parse` with something like `smithy4s#ShapeId$member` you will get `ShapeId("smithy4s", "ShapeId$member")` which is not entirely right

File: modules/core/src/smithy4s/ShapeId.scala [hunk 2]
```diff
@@ -40,12 +40,29 @@ object ShapeId extends ShapeTag.Has[ShapeId] { self =>
     }
   }
 
-  final case class Member(shapeId: ShapeId, member: String)
```
Comment (kubukoz) [inline modules/core/src/smithy4s/ShapeId.scala:43]: Interesting, didn't know we had that. Was this used for anything?

Comment (miguel-vila) [reply modules/core/src/smithy4s/ShapeId.scala:43]: seems like not, there's a `withMember` method and that one is not used either

**Comment by Baccata** [comment]
Sorry for only reviewing this now. Is there a specific usecase for it ? I was tempted to remover the `Member` case class, as we don't use it anywhere

**Comment by Baccata** [comment]
Closing this as it doesn't help with any of the library features, and is simple enough to implement in userland. 

Happy to revisit if there's ever a justification that would warrant it.

**Pull request closed without merging**